MvvmCross/Core/Core/ViewModels/MvxViewModel.cs

```diff
@@ -82,10 +83,56 @@ public abstract class MvxViewModel<TInit> : MvxViewModel, IMvxViewModelInitializ
                 throw new MvxIoCResolveException("There is no implementation of IMvxJsonConverter registered. You need to use the MvvmCross Json plugin or create your own implementation of IMvxJsonConverter.");
             }
 
-            var deserialized = serializer.DeserializeObject<TInit>(parameter);
-            await Init(deserialized);
+            var deserialized = serializer.DeserializeObject<TParameter>(parameter);
+            await Initialize(deserialized);
+        }
+
+        public abstract Task Initialize(TParameter parameter);
+    }
+
+    //TODO: Not possible to name MvxViewModel, name is MvxViewModelResult for now
+    public abstract class MvxViewModelResult<TResult> : MvxViewModel, IMvxViewModelResult<TResult> where TResult : class
+    {
+        TaskCompletionSource<TResult> _tcs;
+
+        public void SetClose(TaskCompletionSource<TResult> tcs)
+        {
+            _tcs = tcs;
```

Comment:
I think we should have a null check here.  What happens if someone calls `SetClose` multiple times?  Does it matter?